### PR TITLE
Allow passing an existing kfold object when creating a new CVGlmNet.

### PR DESCRIPTION
**Pull request by nenorbot**
When evaluating different learning models, it's useful to pass your own KFold object so that they all train and test on the same sets. This change allows the user to pass his own object. If not passed, the behavior is as before.

### CODE REVIEW

**Comment by madrury** [comment]
This is a good idea.  I'll merge in this commit.

There is a minor issue though.  The patch works fine if the user creates an un-weighted cv object by calling unweighted_k_fold of directly instantiating a sklearn.cross_validation.KFold object, but is hindered by the lack of .nfolds and .shuffle attributes if calling weighted_k_fold.  Looks like the *_k_fold functions should return an object.  I'll make it a priority to fix this.
